### PR TITLE
Add file sorting option for TFT SD card and USB drive

### PR DESCRIPTION
**Pull request by guruathwal**
- Add file sorting option for TFT SD card and USB drive. Files can be sorted by Date - Newest first, Date - Oldest first, Name Ascending or Name Descending.
- misc cleanup and bugfix.

Resolves: #601

### CODE REVIEW

File: Copy to SD Card root directory to update/config_rrf.ini [hunk 1]
```diff
@@ -156,11 +156,19 @@ invert_axis:X0 Y0 Z0 LY0
 #   Options: [enable: 1, disable: 0]
 persistent_info:0
 
-#### File List Mode
+#### Files List Mode
 # Display files in list mode instead of icon mode.
 #   Options: [enable: 1, disable: 0]
 files_list_mode:1
 
+#### Files sorting
```
Comment (digant73) [inline Copy to SD Card root directory to update/config_rrf.ini:164]: As for all param titles, use capital letter. so Files Sorting

File: Copy to SD Card root directory to update/config_rrf.ini [hunk 2]
```diff
@@ -156,11 +156,19 @@ invert_axis:X0 Y0 Z0 LY0
...
+# Sort files and folders based on the selected option.
+#
+# NOTE: Only applicable for files in TFT SD Card and TFT USB Stick.
+#
+#   Options: [Date Newest First: 0, Date Oldest First: 1, Name Ascending: 2, Name Descending: 3]
```
Comment (digant73) [inline Copy to SD Card root directory to update/config_rrf.ini:169]: shouldn't better someting like Newest, Oldest, Ascending, Descending? The param name and description are already self explaining

File: TFT/src/User/Fatfs/myfatfs.c [hunk 3]
```diff
@@ -4,6 +4,42 @@
...
+    uint16_t maxlen = (strlen(name1) < strlen(name2)) ? strlen(name1):strlen(name2);
+
+    // compare each character
+    for (uint16_t i = 0; i < maxlen; i++)
+    {
+      // convert all upper case characters to lower case
+      char a = (name1[i] > 64  && name1[i] < 91) ? (name1[i] + 32) : name1[i];
+      char b = (name2[i] > 64  && name2[i] < 91) ? (name2[i] + 32) : name2[i];
+
+      if(a != b)
```
Comment (digant73) [inline TFT/src/User/Fatfs/myfatfs.c:35]: a space after "if"

File: TFT/src/User/Fatfs/myfatfs.c [hunk 4]
```diff
@@ -4,6 +4,42 @@
...
+  // sort by date
+  if (infoSettings.files_sort_by <= SORT_DATE_OLD_FIRST)
+  {
+    // file with most recent date displays first in newest first and last in oldest first
+    return ((date1 > date2) == infoSettings.files_sort_by % 2);
+  }
+  // sort by name
+  else
+  {
+    uint16_t maxlen = (strlen(name1) < strlen(name2)) ? strlen(name1):strlen(name2);
```
Comment (digant73) [inline TFT/src/User/Fatfs/myfatfs.c:26]: a space befor and after ":"

File: TFT/src/User/Menu/FeatureSettings.c [hunk 5]
```diff
@@ -207,6 +221,9 @@ void loadFeatureSettings(LISTITEM * item, uint16_t item_index, uint8_t itemPos)
         item->icon = iconToggle[infoSettings.file_listmode];
         break;
 
+      case SKEY_FILE_SORT_BY:
+        setDynamicTextValue(itemPos, (char *)itemSortBy[infoSettings.files_sort_by]);
+        break;
       case SKEY_ACK_NOTIFICATION:
```
Comment (digant73) [inline TFT/src/User/Menu/FeatureSettings.c:227]: add blank line before "case"

**Review by digant73 — CHANGES_REQUESTED** [review]

File: Copy to SD Card root directory to update/config_rrf.ini (hunk 2 above)
Comment (guruathwal) [reply Copy to SD Card root directory to update/config_rrf.ini:169]: How would you define what is ascending and descending?

**Comment by digant73** [comment]
report "fixes #1901" so that FR will be closed

**Comment by kisslorand** [comment]
> don't forget to report "fixes #1901" so that FR will be closed

It doesn't "fix" that FR.

**Comment by guruathwal** [comment]
@digant73 #901 is not related to this PR.

**Comment by oldman4U** [comment]
But it resolves #601;-) One of the oldest FR's.

**Comment by digant73** [comment]
@guruathwal, @kisslorand yes, sorry my bad

File: Copy to SD Card root directory to update/config_rrf.ini (hunk 2 above)
Comment (digant73) [reply Copy to SD Card root directory to update/config_rrf.ini:169]: talking about standard file names ordering type by date and name. In any case, just my opinion

**Comment by oldman4U** [comment]
Came here to test it and saw that it is already merged;-)

Thank you

**Pull request merged**